In [2]:
# NAME: ATHARVA RANADE
# GITHUB Username: atharvaranade4

# 09-10-2020

In [3]:
#imports

import numpy as np
import pandas as pd

In [4]:
pwd

'C:\\Users\\Owner\\MUSA COURSE\\HW 1'

In [5]:
#import file

Zip_df = pd.read_csv("Zip_Zhvi_AllHomes.csv", encoding = 'latin-1')

In [6]:
Zip_df.head()

RegionID  RegionName      City State                             Metro  \
0     61639       10025  New York    NY       New York-Newark-Jersey City   
1     84654       60657   Chicago    IL          Chicago-Naperville-Elgin   
2     61637       10023  New York    NY       New York-Newark-Jersey City   
3     91982       77494      Katy    TX  Houston-The Woodlands-Sugar Land   
4     84616       60614   Chicago    IL          Chicago-Naperville-Elgin   

        CountyName  SizeRank   1996-04   1996-05   1996-06  ...  2018-02  \
0  New York County         1  171600.0  171600.0  171400.0  ...  1130500   
1      Cook County         2  158400.0  159700.0  160700.0  ...   351600   
2  New York County         3  347900.0  349600.0  351100.0  ...  1516000   
3    Harris County         4  210400.0  212200.0  212200.0  ...   326600   
4      Cook County         5  192500.0  194500.0  196100.0  ...   429000   

   2018-03  2018-04  2018-05  2018-06  2018-07  2018-08  2018-09  2018-10  \
0  1123700  1119500  1116900  1110100  1098400  1086900  1080500  1072200   
1   352900   351900   350400   348700   347800   348200   349500   351500   
2  1497900  1497800  1504600  1489900  1463300  1438800  1411600  1389900   
3   330400   332600   334500   335800   336900   338200   338400   336900   
4   430400   429500   428600   428700   430600   431900   430900   430900   

   2018-11  
0  1064000  
1   354000  
2  1380100  
3   336000  
4   433200  

[5 rows x 279 columns]

In [10]:
#List of cities we want

county_name = ["Philadelphia County"]
#Zip_df["City"].isin(cities_list)

Zip_df = Zip_df.loc[Zip_df["CountyName"].isin(county_name)]
Zip_df.head()

RegionID  RegionName          City State                           Metro  \
214     65779       19111  Philadelphia    PA  Philadelphia-Camden-Wilmington   
300     65791       19124  Philadelphia    PA  Philadelphia-Camden-Wilmington   
377     65787       19120  Philadelphia    PA  Philadelphia-Camden-Wilmington   
542     65815       19148  Philadelphia    PA  Philadelphia-Camden-Wilmington   
690     65812       19145  Philadelphia    PA  Philadelphia-Camden-Wilmington   

              CountyName  SizeRank  1996-04  1996-05  1996-06  ...  2018-02  \
214  Philadelphia County       215  84900.0  84700.0  84500.0  ...   174300   
300  Philadelphia County       301  43100.0  43000.0  42900.0  ...    86200   
377  Philadelphia County       378  46100.0  46100.0  46100.0  ...    91100   
542  Philadelphia County       543  41100.0  41100.0  41000.0  ...   205700   
690  Philadelphia County       691  41000.0  41000.0  41000.0  ...   204100   

     2018-03  2018-04  2018-05  2018-06  2018-07  2018-08  2018-09  2018-10  \
214   175500   175700   175800   176400   177300   178400   180300   182800   
300    88000    88900    89900    90800    91400    91900    92800    94200   
377    92100    92400    92400    92600    92800    92800    92900    93500   
542   209500   211800   213400   214900   215500   215700   216900   219800   
690   208400   209500   209100   209600   211100   212600   213800   215100   

     2018-11  
214   184800  
300    95700  
377    94200  
542   222500  
690   216700  

[5 rows x 279 columns]

In [8]:
#Drop unwanted columns
#reset index to 0

drop_column = ["State" , "RegionID" , "Metro" , "City" , "CountyName" , "SizeRank"]
Zip_trimmed = Zip_df.drop(drop_column, axis = 1)

Zip_trimmed
home_values = Zip_trimmed.reset_index(drop = True)
home_values.head()

RegionName  1996-04  1996-05  1996-06  1996-07  1996-08  1996-09  1996-10  \
0       19111  84900.0  84700.0  84500.0  84300.0  84200.0  84100.0  83900.0   
1       19124  43100.0  43000.0  42900.0  42700.0  42500.0  42400.0  42300.0   
2       19120  46100.0  46100.0  46100.0  46100.0  46000.0  45900.0  45900.0   
3       19148  41100.0  41100.0  41000.0  40900.0  40700.0  40600.0  40500.0   
4       19145  41000.0  41000.0  41000.0  40900.0  40800.0  40700.0  40700.0   

   1996-11  1996-12  ...  2018-02  2018-03  2018-04  2018-05  2018-06  \
0  83800.0  83700.0  ...   174300   175500   175700   175800   176400   
1  42300.0  42300.0  ...    86200    88000    88900    89900    90800   
2  45800.0  45700.0  ...    91100    92100    92400    92400    92600   
3  40400.0  40400.0  ...   205700   209500   211800   213400   214900   
4  40800.0  40900.0  ...   204100   208400   209500   209100   209600   

   2018-07  2018-08  2018-09  2018-10  2018-11  
0   177300   178400   180300   182800   184800  
1    91400    91900    92800    94200    95700  
2    92800    92800    92900    93500    94200  
3   215500   215700   216900   219800   222500  
4   211100   212600   213800   215100   216700  

[5 rows x 273 columns]

In [8]:
#Melting data to make it in tidy vertcial format

ZHVI = pd.melt(
    home_values, id_vars=["RegionName"], value_name="ZHVI", var_name="Date"
)
ZHVI

RegionName     Date      ZHVI
0          19111  1996-04   84900.0
1          19124  1996-04   43100.0
2          19120  1996-04   46100.0
3          19148  1996-04   41100.0
4          19145  1996-04   41000.0
...          ...      ...       ...
9787       19129  2018-11  230000.0
9788       19118  2018-11  559500.0
9789       19102  2018-11  361400.0
9790       19127  2018-11  224900.0
9791       19137  2018-11  162200.0

[9792 rows x 3 columns]

In [9]:
# convert to timestamp
ZHVI['Date'] = pd.to_datetime(ZHVI['Date'])

In [10]:
ZHVI['Month'] = ZHVI['Date'].dt.month
ZHVI['Year'] = ZHVI['Date'].dt.year

In [11]:
ZHVI

RegionName       Date      ZHVI  Month  Year
0          19111 1996-04-01   84900.0      4  1996
1          19124 1996-04-01   43100.0      4  1996
2          19120 1996-04-01   46100.0      4  1996
3          19148 1996-04-01   41100.0      4  1996
4          19145 1996-04-01   41000.0      4  1996
...          ...        ...       ...    ...   ...
9787       19129 2018-11-01  230000.0     11  2018
9788       19118 2018-11-01  559500.0     11  2018
9789       19102 2018-11-01  361400.0     11  2018
9790       19127 2018-11-01  224900.0     11  2018
9791       19137 2018-11-01  162200.0     11  2018

[9792 rows x 5 columns]

In [12]:
annual_ZHVI = ZHVI.groupby(["RegionName", "Year"])["ZHVI"].mean()
annual_ZHVI.head()

RegionName  Year
19102       1996     79966.666667
            1997     83166.666667
            1998     92550.000000
            1999    114358.333333
            2000    145175.000000
Name: ZHVI, dtype: float64

In [13]:
annual_ZHVI = annual_ZHVI.reset_index()

In [14]:
annual_ZHVI

RegionName  Year           ZHVI
0         19102  1996   79966.666667
1         19102  1997   83166.666667
2         19102  1998   92550.000000
3         19102  1999  114358.333333
4         19102  2000  145175.000000
..          ...   ...            ...
823       19154  2014  174100.000000
824       19154  2015  171233.333333
825       19154  2016  186883.333333
826       19154  2017  198741.666667
827       19154  2018  211436.363636

[828 rows x 3 columns]

In [18]:
#sorting the ZHVI value in descending order

annual_ZHVI_sorted = annual_ZHVI.sort_values(by = ["ZHVI"] , ascending = False)
annual_ZHVI_sorted

RegionName  Year           ZHVI
206       19118  2018  535990.909091
45        19103  2018  523754.545455
205       19118  2017  520166.666667
44        19103  2017  516750.000000
204       19118  2016  499550.000000
..          ...   ...            ...
255       19122  1998            NaN
256       19122  1999            NaN
257       19122  2000            NaN
258       19122  2001            NaN
259       19122  2002            NaN

[828 rows x 3 columns]

In [22]:
#Display final answer:

(annual_ZHVI_sorted.loc[[206]])

RegionName  Year           ZHVI
206       19118  2018  535990.909091

#### FINAL ANSWER:

ZIP CODE 19118 has the maximum value of 535990.9 in the year 2018